? rsc/.2.html
? rsc/doc++
? rsc/rsc-list
? rsc/test.rc
? rsc/test.src
? rsc/inc/rsclayout.hxx
? rsc/source/res/rsclayout.cxx
Index: rsc/inc/rscdb.hxx
===================================================================
RCS file: /cvs/gsl/rsc/inc/rscdb.hxx,v
retrieving revision 1.8
diff -u -p -u -r1.8 rscdb.hxx
--- rsc/inc/rscdb.hxx	2 Jul 2002 14:25:56 -0000	1.8
+++ rsc/inc/rscdb.hxx	22 Aug 2003 17:45:55 -0000
@@ -226,6 +226,8 @@ class RscTypCont
 								RscTop * pClassColor );
 	RscTop *	InitClassImageList( RscTop * pSuper, RscTop *pClassBitmap,
 									RscTop * pClassColor );
+
+	RscTop *	InitClassLayout( RscTop * pSuper );
 	RscTop *	InitClassWindow( RscTop * pSuper, RscEnum * pMapUnit,
 								 RscArray * pLangGeo );
 	RscTop *	InitClassSystemWindow( RscTop * pSuper );
Index: rsc/source/parser/erscerr.cxx
===================================================================
RCS file: /cvs/gsl/rsc/source/parser/erscerr.cxx,v
retrieving revision 1.6
diff -u -p -u -r1.6 erscerr.cxx
--- rsc/source/parser/erscerr.cxx	7 Nov 2001 16:51:18 -0000	1.6
+++ rsc/source/parser/erscerr.cxx	22 Aug 2003 17:45:55 -0000
@@ -464,7 +464,7 @@ void RscError::ErrorFormat( const ERRTYP
 		StdLstErr( "\n" );
 	}
 	StdLstErr( "f" );
-	sprintf( buf, "%u", (USHORT)rError );
+	sprintf( buf, "0x%x", (USHORT)rError );
 	StdLstErr( buf );
 
 	if( pFI && pTC ){
Index: rsc/source/parser/rscicpx.cxx
===================================================================
RCS file: /cvs/gsl/rsc/source/parser/rscicpx.cxx,v
retrieving revision 1.6
diff -u -p -u -r1.6 rscicpx.cxx
--- rsc/source/parser/rscicpx.cxx	26 Mar 2003 15:50:46 -0000	1.6
+++ rsc/source/parser/rscicpx.cxx	22 Aug 2003 17:45:55 -0000
@@ -281,6 +281,32 @@ RscTop * RscTypCont::InitClassImageList(
 	return( pClassImageList );
 }
 
+RscTop * RscTypCont::InitClassLayout( RscTop * pSuper )
+{
+	HASHID  nId;
+	RscTop *pClassLayout;
+
+	// Layout
+	nId = pHS->Insert( "Layout" );
+//	pClassLayout = new RscLayout( nId, RSC_LAYOUT, pSuper);
+	pClassLayout = new RscClass( nId, RSC_LAYOUT, pSuper );
+	aNmTb.Put( nId, CLASSNAME, pClassLayout );
+
+	nId = aNmTb.Put( "PeerId", VARNAME );
+	pClassLayout->SetVariable( nId, &aIdNoZeroUShort, NULL, 0, RSC_LAYOUT_PEER );
+
+	nId = aNmTb.Put( "Expand", VARNAME );
+	pClassLayout->SetVariable( nId, &aBool, NULL, 0, RSC_LAYOUT_EXPAND );
+	nId = aNmTb.Put( "Fill", VARNAME );
+	pClassLayout->SetVariable( nId, &aBool, NULL, 0, RSC_LAYOUT_FILL );
+	nId = aNmTb.Put( "Homogeneous", VARNAME );
+	pClassLayout->SetVariable( nId, &aBool, NULL, 0, RSC_LAYOUT_HOMOGENEOUS );
+
+	pRoot->Insert( pClassLayout );
+
+	return pClassLayout;
+}
+
 /*************************************************************************
 |*	  RscTypCont::InitClassWindow()
 *************************************************************************/
@@ -315,7 +341,7 @@ RscTop * RscTypCont::InitClassWindow( Rs
 		// Variable einfuegen
 		nVarId = aNmTb.Put( "_RscExtraFlags", VARNAME );
 		pClassWindow->SetVariable( nVarId, pFlag, NULL,
-																		VAR_HIDDEN | VAR_NOENUM );
+					   VAR_HIDDEN | VAR_NOENUM );
 
 		aBaseLst.Insert(
 			pClient = new RscClient( pHS->Insert( "BOOL" ), RSC_NOTYPE,
@@ -373,6 +399,8 @@ RscTop * RscTypCont::InitClassWindow( Rs
 	pClassWindow->SetVariable( nId, &aLong, NULL, 0, WINDOW_EXTRALONG );
 	nId = aNmTb.Put( "UniqueId", VARNAME );
 	pClassWindow->SetVariable( nId, &aLong, NULL, 0, WINDOW_UNIQUEID );
+	nId = aNmTb.Put( "LayoutId", VARNAME );
+	pClassWindow->SetVariable( nId, &aIdNoZeroUShort, NULL, 0, WINDOW_LAYOUT );
 
 	return( pClassWindow );
 }
Index: rsc/source/parser/rscinit.cxx
===================================================================
RCS file: /cvs/gsl/rsc/source/parser/rscinit.cxx,v
retrieving revision 1.7
diff -u -p -u -r1.7 rscinit.cxx
--- rsc/source/parser/rscinit.cxx	2 Jul 2002 14:44:00 -0000	1.7
+++ rsc/source/parser/rscinit.cxx	22 Aug 2003 17:45:55 -0000
@@ -517,6 +517,8 @@ void RscTypCont::Init()
 	pClassWindow = InitClassWindow( pClassMgr, pMapUnit,
 									pLangGeometry );
 	pRoot->Insert( pClassWindow );
+
+	InitClassLayout( pClassMgr );
 }
 {
 
Index: rsc/source/res/rsccont.cxx
===================================================================
RCS file: /cvs/gsl/rsc/source/res/rsccont.cxx,v
retrieving revision 1.2
diff -u -p -u -r1.2 rsccont.cxx
--- rsc/source/res/rsccont.cxx	10 Oct 2001 11:51:25 -0000	1.2
+++ rsc/source/res/rsccont.cxx	22 Aug 2003 17:45:55 -0000
@@ -292,14 +292,15 @@ RSCINST RscBaseCont::SearchEle( const RS
 |*
 *************************************************************************/
 ERRTYPE RscBaseCont::GetElement( const RSCINST & rInst, const RscId & rEleName,
-								 RscTop * pCreateClass,
-								 const RSCINST & rCreateInst, RSCINST * pGetInst )
+				 RscTop * pCreateClass,
+				 const RSCINST & rCreateInst, RSCINST * pGetInst )
 {
 	RscBaseContInst * pClassData;
 	RSCINST 		  aTmpI;
 	ERRTYPE 		  aError;
 
-	if( !bNoId && !rEleName.IsId() )
+	if( !bNoId && !rEleName.IsId() &&
+	    pCreateClass->GetTypId() != RSC_LAYOUT )
 		aError = WRN_CONT_NOID;
 	pClassData = (RscBaseContInst *)(rInst.pData + nOffInstData);
 
